### ISSUE TITLE
An unhandled exception occured

### ISSUE THREAD

**Issue by Predatorie** [softsan/HorizontalListView#1]
Hi, I am trying to implement the control in a Xamarin Forms project using VS2017RC and a crash is occurring when I attempt to assign an ItemsSource to the control. The ItemsSource is a List of HamburgerMenuItems that contain three public text properties as shown below.

If I remove the ItemsSource assignment the control will be displayed in the page once it has loaded on the Android emulator. Is it possible I am missing a step here as I am relatively new to Xamarin development.

![menu](https://cloud.githubusercontent.com/assets/2666052/23268740/9b84c76a-f9bc-11e6-83d4-7ea69356c72f.png)

![menu_list](https://cloud.githubusercontent.com/assets/2666052/23268969/19370236-f9bd-11e6-8dd2-5e18c6bcd09a.png)

![xaml](https://cloud.githubusercontent.com/assets/2666052/23268789/b4d248be-f9bc-11e6-8493-fd32bd48b398.png)

**Comment by softsan** [comment]
The `ItemsSource ` binding should be an `ObservableCollection`, not a simple list. and should be inherited from INotifyPropertyChanged / OnPropertyChanged. Please see "CategoryViewModel" under "ViewModel" folder in the sample.

**Comment by Predatorie** [comment]
I see, I was a little baffled because swapping out the control for a regular Listview control worked. INotifyPropertyChange is implemented via Prism. Appreciate the quick reply and the information thank you.

**Comment by softsan** [comment]
no worries. Does it resolved your problem?

**Comment by Predatorie** [comment]
I had finished for the day but will jump on it first thing and let you know.

**Comment by Predatorie** [comment]
I've spent a little time on this but as I suspected switching from a plain List to an ObservableCollection makes no difference. Digging a little deeper I believe the control may not work correctly under VS2017RC2 that I am using and/or the Xamarin.Forms version I am using 2.3.1.114 targeting monoandroid71. I am unable to successfully build your project and unfortunately I don't have a lot of time to look more into this as I am on a very aggressive schedule so I may have to seek an alternate. In any event thank you for you quick replies and assistance. I will watch this project in the event any updates. Thanks again.
